sentry/src/main/java/io/sentry/Span.java

```diff
@@ -72,15 +88,21 @@ public Span(
     this.context = Objects.requireNonNull(context, "context is required");
     this.transaction = Objects.requireNonNull(sentryTracer, "sentryTracer is required");
     this.hub = Objects.requireNonNull(hub, "hub is required");
-    this.startTimestamp = startTimestamp != null ? startTimestamp : DateUtils.getCurrentDateTime();
     this.spanFinishedCallback = null;
+    if (startTimestamp != null) {
+      this.startTimestamp = startTimestamp;
+      this.startNanos = null;
+    } else {
+      this.startTimestamp = DateUtils.getCurrentDateTime();
+      this.startNanos = System.nanoTime();
+    }
   }
 
   public @NotNull Date getStartTimestamp() {
     return startTimestamp;
   }
 
-  public @Nullable Date getTimestamp() {
+  public @Nullable Double getTimestamp() {
```

Comment:
Is there a way to make this change without a breaking change?
Like adding another `timestampHigherPrecision()` and leave this one as `Date timestamp()` and have it return a `Date` from the float for example?

We'd need to bump a major to release this change

Reply:
Definitely there is a way, the question is if it's worth it considering this class is marked as internal. `getTimestamp` is never used by the user code as it's not exposed with `ISpan`.